### DISCUSSION TITLE
Where should test results and coverage be written

### DISCUSSION THREAD

**Discussion by bradleyjames** [goreleaser/goreleaser#5306] [Ideas, Issue Triage, and General discussions]
In previous versions of goreleaser I wrote test output to the dist directory. 

.goreleaser.yaml
```
before:
  hooks:
    - make test
```

Makefile
```
test:
	mkdir -p dist
	go test -v ./... -cover -coverprofile dist/cover.out
	go tool cover -html dist/cover.out -o dist/cover.html
```

Now when I run `goreleaser build --single-target --snapshot --clean` my build is failing with:

```  
⨯ build failed after 4s                    error=dist is not empty, remove it before running goreleaser or use the --clean flag
```

Where should I be writing the test results so that they're available for traceability while building my go binary as well as failing my build if the tests don't pass?  I feel like I'm missing a basic convention that has been updated.

**Comment by caarlos0** [comment]
Hey!

Since very early versions this is not allowed, not sure how long ago this worked.

Anyways, I would recommend running tests outside of goreleaser. 

GoReleaser is meant to be used **within** a CI, not **as** a CI.

Alternatively, if you really want to do that, you can gitignore `cover.out` and `cover.html` and write the the repo's root dir.
